chia/simulator/block_tools.py

```diff
@@ -358,26 +358,34 @@ async def setup_plots(
         num_non_keychain_plots: int = 3,
         plot_size: int = 20,
         bitfield: bool = True,
-    ):
+    ) -> bool:
         self.add_plot_directory(self.plot_dir)
         assert self.created_plots == 0
+        existing_plots: bool = True
         # OG Plots
         for i in range(num_og_plots):
-            await self.new_plot(plot_size=plot_size, bitfield=bitfield)
+            _, plot_exists = await self.new_plot(plot_size=plot_size, bitfield=bitfield)
+            if existing_plots and not plot_exists:
```

Comment:
I think resetting to `False` again is fine and simplifies the pattern.  A couple repetitions below.

```suggestion
            if not plot_exists:
```

Reply:
fair point, not sure why i added that.